lib/streamlit/script_request_queue.py

```diff
@@ -32,32 +30,31 @@ class ScriptRequest(Enum):
     SHUTDOWN = "SHUTDOWN"
 
 
-@attr.s(auto_attribs=True, slots=True)
-class RerunData:
-    """Data attached to RERUN requests."""
+class RerunData(NamedTuple):
+    """Data attached to RERUN requests. Immutable."""
 
     query_string: str = ""
     widget_states: Optional[WidgetStates] = None
 
 
-@attr.s(auto_attribs=True, slots=True)
```

Comment:
Is the idea that using attrs for a non-dataclass is less clear than making it a regular python class?

Reply:
Yes - I think when someone sees `@attr.s` their first assumption is, "this is a dataclass", but ScriptRequestQueue has nothing very dataclass-like about it. It feels like a semantic misdirection.